### PR TITLE
Gatsby remark images default alt tags and optional linking

### PR DESCRIPTION
**Pull request by chiedo**
Feature 1:

- Now alt tags will be automatically generated from the file name if an alt tag is not present.

Feature 2:

- Images being links are now optional

### CODE REVIEW

File: packages/gatsby-remark-images/README.md [hunk 1]
```diff
@@ -33,6 +33,9 @@ plugins: [
             // the content container as this plugin uses this as the
             // base for generating different widths of each image.
             maxWidth: 590,
+            // Remove the default behavior of adding a link to each
+            // image.
+            linkImages: false,
```
Comment (KyleAMathews) [inline packages/gatsby-remark-images/README.md]: could you make this `linkImagesToOriginal`

File: packages/gatsby-remark-images/src/index.js [hunk 2]
```diff
@@ -83,43 +84,54 @@ module.exports = (
               const fallbackSrc = responsiveSizesResult.src
               const srcSet = responsiveSizesResult.srcSet
 
+              // Generate default alt tag
+              const srcSplit = node.url.split("/");
+              const fileName = srcSplit[srcSplit.length - 1];
+              const fileNameNoExt = fileName.replace(/\.[^/.]+$/, "")
+              const defaultAlt = fileNameNoExt.replace(/[^A-Z0-9]/ig, " ");
```
Comment (KyleAMathews) [inline packages/gatsby-remark-images/src/index.js]: 🔥

File: packages/gatsby-remark-images/README.md (hunk 1 above)
Comment (chiedo) [inline packages/gatsby-remark-images/README.md]: Done!

File: packages/gatsby-remark-images/src/index.js (hunk 2 above)
Comment (chiedo) [inline packages/gatsby-remark-images/src/index.js]: 😄

**Comment by KyleAMathews** [comment]
Thanks!

**Pull request merged**